--- a/src/backend/INotify.zig
+++ b/src/backend/INotify.zig
@@ -141,9 +141,10 @@
         pub fn add_watch(self: *@This(), allocator: std.mem.Allocator, path: []const u8) error{ OutOfMemory, WatchFailed }!void {
             const path_z = try allocator.dupeZ(u8, path);
             defer allocator.free(path_z);
-            const wd = std.os.linux.inotify_add_watch(self.inotify_fd, path_z, watch_mask);
-            switch (std.posix.errno(wd)) {
-                .SUCCESS => {},
+            const wd_unsigned = std.os.linux.inotify_add_watch(self.inotify_fd, path_z, watch_mask);
+            const wd_signed: isize = @bitCast(wd_unsigned);
+            switch (wd_signed < 0) {
+                true => {
                     std.log.err("nightwatch.add_watch failed: {t}", .{e});
                     return error.WatchFailed;
                 },
